### DISCUSSION TITLE
Problem with detected db_name

### DISCUSSION THREAD

**Discussion by mlawson00** [mikro-orm/mikro-orm#5608] [General]
Hi! 👋 
      
Firstly, thanks for your work on this project! 🙂

Today I used [patch-package](https://github.com/ds300/patch-package) to patch `@mikro-orm/core@5.9.7` for the Medusa-js project I'm working on.

I had some issues with the database name extracted from the db_url of my database. I am using cloud run, which makes google cloud sql a necessity. It can only be connected to using unix sockets, which means the dbname is no longer the last argument used. For example, my connection url looks like:

postgresql://USER:PWD@localhost/DBNAME?host=/cloudsql/PROJECT:REGION:INSTANCE

Here is the diff that solved my problem:

```diff
diff --git a/node_modules/@mikro-orm/core/utils/Configuration.js b/node_modules/@mikro-orm/core/utils/Configuration.js
index 1c02496..9824e0d 100644
--- a/node_modules/@mikro-orm/core/utils/Configuration.js
+++ b/node_modules/@mikro-orm/core/utils/Configuration.js
@@ -168,7 +168,7 @@ class Configuration {
         }
         const url = this.getClientUrl().match(/:\/\/.*\/([^?]+)/);
         if (url) {
-            this.options.dbName = this.get('dbName', decodeURIComponent(url[1]));
+            this.options.dbName = this.get('dbName', process.env.DATABASE_DATABASE || url[1]);
         }
         if (!this.options.charset) {
             this.options.charset = this.platform.getDefaultCharset();
```

<em>This issue body was [partially generated by patch-package](https://github.com/ds300/patch-package/issues/296).</em>

**Comment by B4nan** [comment]
The second parameter for the `get` method is the default value, instead of changing how the default is inferred, you could simply provide the value explicitly, either via `dbName` option in the config or via `MIKRO_ORM_DB_NAME` env var.

**Reply by B4nan** [reply]
This should help to make it work out of box: 1472705bc1e974b09aa00a495b79f07cb776f5b8
